tokio/src/runtime/metrics/histogram.rs

```diff
@@ -112,52 +224,62 @@ impl HistogramBatch {
     }
 
     fn value_to_bucket(&self, value: u64) -> usize {
-        match self.scale {
-            HistogramScale::Linear => {
-                let max = self.buckets.len() - 1;
-                cmp::min(value / self.resolution, max as u64) as usize
-            }
-            HistogramScale::Log => {
-                let max = self.buckets.len() - 1;
-
-                if value < self.resolution {
-                    0
-                } else {
-                    let significant_digits = 64 - value.leading_zeros();
-                    let bucket_digits = 64 - (self.resolution - 1).leading_zeros();
-                    cmp::min(significant_digits as usize - bucket_digits as usize, max)
-                }
-            }
-        }
+        self.configuration.value_to_bucket(value)
     }
 }
 
 impl HistogramBuilder {
     pub(crate) fn new() -> HistogramBuilder {
         HistogramBuilder {
-            scale: HistogramScale::Linear,
-            // Resolution is in nanoseconds.
-            resolution: 100_000,
-            num_buckets: 10,
+            histogram_type: HistogramType::Linear(LinearHistogram {
+                num_buckets: 10,
+                bucket_width: 100_000,
+            }),
+            legacy: None,
+        }
+    }
+
+    pub(crate) fn legacy_mut(&mut self, f: impl Fn(&mut LegacyBuilder)) {
+        if let Some(legacy) = &mut self.legacy {
+            f(legacy)
+        } else {
+            let mut legacy = LegacyBuilder::default();
+            f(&mut legacy);
+            self.legacy = Some(legacy)
         }
```

Comment:
`Option` has a function that does handles the None case for you:

```suggestion
        let legacy = self.legacy.get_or_insert_with(|| LegacyBuilder::default());
        f(legacy);
```